### PR TITLE
feat: new info panel

### PR DESCRIPTION
**Pull request by kaloster**
#947 

UI updated to accommodate side by side for spatial:
- Removed dataset drawer
- Moved dataset info to new info panel
- Moved gene info to new info panel
- New panel can be minimized/maximized and closed
- Revised and fixed tests

Cleaned up unused code and redux actions

https://github.com/chanzuckerberg/single-cell-explorer/assets/7976579/11b62b25-bfbb-457f-99f2-18e73fae0105

### CODE REVIEW

File: client/src/components/geneExpression/infoPanel/index.tsx [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+import DatasetInfo from "./datasetInfo";
+
+interface Props {
+  activeTab: string;
+  dispatch: AppDispatch;
+  infoPanelMinimized: boolean;
+  infoPanelHidden: boolean;
+}
+
+const InfoPanel = (props: Props) => {
```
Comment (tihuan) [inline client/src/components/geneExpression/infoPanel/index.tsx:23]: super nit: Can we change this from function expression to function declaration please 🙏 ? I believe stack tracing works a lot better with declaration than an anonymous function!

Here and throughout

E.g.,

```ts
function InfoPanel(props: Props) {
  ...
}
```

File: client/src/components/geneExpression/infoPanel/index.tsx [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+  InfoPanelContent,
+  InfoPanelHeader,
+  InfoPanelTabs,
+  InfoPanelWrapper,
+  StyledAnchorButton,
+} from "./style";
+import GeneInfo from "./geneInfo";
+import DatasetInfo from "./datasetInfo";
+
+interface Props {
```
Comment (tihuan) [inline client/src/components/geneExpression/infoPanel/index.tsx:16]: Non-blocking: Maybe as a post launch P0 - worth refactoring types out to its own file like we do in Data Portal 🙆‍♂️ ?

File: client/src/components/geneExpression/infoPanel/geneInfo/index.tsx [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+                  display: "-webkit-box",
+                  WebkitLineClamp: "7",
+                  WebkitBoxOrient: "vertical",
+                  overflow: "hidden",
+                }}
+                data-testid="gene-info-summary"
+              >
+                {geneSummary}
+              </Content>
+            )}
```
Comment (tihuan) [inline client/src/components/geneExpression/infoPanel/geneInfo/index.tsx:135]: wonder if there's opportunities to keeping the code DRY by reusing the same styling object?

File: client/src/components/geneExpression/infoPanel/geneInfo/index.tsx [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+                  sdsType="static"
+                />
+                <span>
+                  NCBI didn&apos;t return an exact match for this gene.
+                </span>
+              </WarningBanner>
+            ) : null}
+            <GeneSymbol data-testid="gene-info-symbol">{gene}</GeneSymbol>
+            <Content data-testid="gene-info-name">{geneName}</Content>
+            {geneSummary === "" ? (
```
Comment (tihuan) [inline client/src/components/geneExpression/infoPanel/geneInfo/index.tsx:112]: non blocking, but something worth refactoring post launch - wonder if we should extract this component out? This render function has so many conditional and nested rendering that the cognitive complexity seems high to me 😁

File: client/src/reducers/controls.ts [hunk 5]
```diff
@@ -1,63 +1,14 @@
 import { AnyAction } from "redux";
 
-type Level = "top" | "bottom" | "";
-
-interface StackLevels {
-  geneLevel: Level;
-  scatterplotLevel: Level;
-}
-/* logic for minimizing and maximizing pop-ups */
-const minimizeMaximizePopUps = (
```
Comment (tihuan) [inline client/src/reducers/controls.ts:10]: really awesome cleanup! Thank you!

File: client/src/components/geneExpression/gene.tsx [hunk 6]
```diff
@@ -91,41 +91,27 @@ class Gene extends React.Component<Props, State> {
...
     const info = await actions.fetchGeneInfo(geneId, gene);
     if (!info) {
-      dispatch({
-        type: "open gene info",
-        gene,
-        url: "",
-        name: "",
-        synonyms: [],
-        summary: "",
-        infoError: "fetch gene info failed",
```
Comment (tihuan) [inline client/src/components/geneExpression/gene.tsx:120]: maybe I'm missing something, but it seems like by removing this dispatch, we no longer tell user a gene info fetch as failed?

File: client/src/components/geneExpression/infoPanel/datasetInfo/index.tsx [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+  allCategoryNames.forEach((catName) => {
+    const isUserAnno = schema?.annotations?.obsByName[catName]?.writable;
+    const colSchema = schema.annotations.obsByName[catName];
+    if (!isUserAnno && colSchema.categories?.length === 1) {
+      allSingleValues.set(catName, colSchema.categories[0]);
+    }
+  });
+  singleContinuousValues.forEach((value, catName) => {
+    allSingleValues.set(catName, value);
+  });
```
Comment (tihuan) [inline client/src/components/geneExpression/infoPanel/datasetInfo/index.tsx:38]: non blocking - just wondering if there's an opportunity to cache the calculation, so we don't generate new objects every time, which could cause unnecessary re-rendering. But up to you!

File: client/src/components/geneExpression/infoPanel/index.tsx [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+        isMinimized={infoPanelMinimized}
+      >
+        {activeTab === "Gene" && <GeneInfo />}
+        {activeTab === "Dataset" && <DatasetInfo />}
+      </InfoPanelContent>
+    </InfoPanelWrapper>
+  );
+};
+
+export default connect((state: RootState) => ({
```
Comment (tihuan) [inline client/src/components/geneExpression/infoPanel/index.tsx:99]: in other places we define and use `mapStateToProps`, maybe worth doing the same here for consistency?

**Review by tihuan — COMMENTED** [review]
Thanks for polishing and cleaning up the components in addition to adding the new side panel feature 🔥 !! Just some small comments PTAL thank you 👼 🙏 ✨

File: client/src/components/geneExpression/infoPanel/geneInfo/index.tsx (hunk 4 above)
Comment (kaloster) [reply client/src/components/geneExpression/infoPanel/geneInfo/index.tsx:112]: I'll address this in a following PR

File: client/src/components/geneExpression/gene.tsx (hunk 6 above)
Comment (kaloster) [reply client/src/components/geneExpression/gene.tsx:120]: this was kind of a workaround to trigger empty state in the UI - no longer needed with the new revised logic introduced in `geneInfo/index.tsx`

"open gene info" will still return `infoError` if it occurs and will affect the UI representation accordingly.

File: client/src/components/geneExpression/index.tsx [hunk 9]
```diff
@@ -175,7 +176,14 @@ class GeneExpression extends React.Component<{}, State> {
     const { isCellGuideCxg } = this.props;
     const { geneSetsExpanded, markerGeneSetsExpanded } = this.state;
     return (
-      <div>
+      <div
+        style={{
+          display: "flex",
+          flexDirection: "column",
+          justifyContent: "space-between",
+          padding: globals.leftSidebarSectionPadding,
```
Comment (seve) [inline client/src/components/geneExpression/index.tsx:184]: we're using a lsb global on the rsb, should we be using a different one or normailize this global?

File: client/src/components/leftSidebar/types.ts [hunk 10]
```diff
@@ -0,0 +1,11 @@
+import { RootState } from "../../reducers";
+
+export interface Props {
+  scatterplotXXaccessor: string;
+  scatterplotYYaccessor: string;
```
Comment (seve) [inline client/src/components/leftSidebar/types.ts:5]: Let's grab these straight from the store! YMMV on my suggestion
```suggestion
  scatterplotXXaccessor: RootState["controls"]["scatterployXXaccessor"];
  scatterplotYYaccessor:  RootState["controls"]["scatterployYYaccessor"];
```

**Review by seve — CHANGES_REQUESTED** [review]
There's also an open conversation in Slack regarding rebasing this branch on the featture branch for side by side

File: client/src/components/geneExpression/gene.tsx [hunk 11]
```diff
@@ -91,41 +91,27 @@ class Gene extends React.Component<Props, State> {
...
-      });
-      return;
-    }
-
     dispatch({
       type: "load gene info",
       gene,
     });
 
+    dispatch({ type: "toggle active info panel", activeTab: "Gene" });
```
[the remark this answers is not in the document]
Comment (kaloster) [reply client/src/components/geneExpression/gene.tsx:104]: Done!

**Comment by kaloster** [comment]
rebased and merged [here](https://github.com/chanzuckerberg/single-cell-explorer/pull/972)

**Pull request closed without merging**